lib/core/src/signer.rs

```diff
@@ -0,0 +1,455 @@
+            .try_into()
+            .map_err(|_| {
+                SignError::Generic(anyhow::anyhow!("Wrong slip77 master blinding key length"))
+            })?;
+        Ok(bytes.into())
+    }
+
+    fn derive_xpub(&self, path: &DerivationPath) -> Result<Xpub, Self::Error> {
+        let pubkey_bytes = self.sdk_signer.derive_xpub(path.to_string())?;
+        let xpub = Xpub::decode(pubkey_bytes.as_slice())?;
+        Ok(xpub)
+    }
+}
+
+pub struct SdkSigner {
+    xprv: Xpriv,
+    secp: Secp256k1<All>, // could be sign only, but it is likely the caller already has the All context.
+    mnemonic: Mnemonic,
+    network: Network,
+}
+
+impl SdkSigner {
+    pub fn new(mnemonic: &str, is_mainnet: bool) -> Result<Self, NewError> {
+        let secp = Secp256k1::new();
+        let mnemonic: Mnemonic = mnemonic.parse()?;
+        let seed = mnemonic.to_seed("");
+
+        let network = if is_mainnet {
+            bitcoin::Network::Bitcoin
+        } else {
+            bitcoin::Network::Testnet
+        };
+
+        let xprv = Xpriv::new_master(network, &seed)?;
+
+        Ok(Self {
+            xprv,
+            secp,
+            mnemonic,
+            network,
+        })
+    }
+
+    fn seed(&self) -> [u8; 64] {
+        self.mnemonic.to_seed("")
+    }
+}
+
+impl Signer for SdkSigner {
+    fn xpub(&self) -> Result<Vec<u8>, SignerError> {
+        Ok(Xpub::from_priv(&self.secp, &self.xprv).encode().to_vec())
+    }
+
+    fn derive_xpub(&self, derivation_path: String) -> Result<Vec<u8>, SignerError> {
+        let der: DerivationPath = derivation_path.parse()?;
+        let derived = self.xprv.derive_priv(&self.secp, &der)?;
+        Ok(Xpub::from_priv(&self.secp, &derived).encode().to_vec())
+    }
+
+    fn sign_ecdsa(&self, msg: Vec<u8>, derivation_path: String) -> Result<Vec<u8>, SignerError> {
+        let der: DerivationPath = derivation_path.parse()?;
+        let ext_derived = self.xprv.derive_priv(&self.secp, &der)?;
+        let sig = self.secp.sign_ecdsa_low_r(
+            &Message::from_digest(
+                msg.try_into()
+                    .map_err(|_| anyhow::anyhow!("failed to sign"))?,
+            ),
+            &ext_derived.private_key,
+        );
+        Ok(sig.serialize_der().to_vec())
+    }
+
+    fn slip77_master_blinding_key(&self) -> Result<Vec<u8>, SignerError> {
+        let seed = self.seed();
+        let master_blinding_key = MasterBlindingKey::from_seed(&seed[..]);
+        Ok(master_blinding_key.as_bytes().to_vec())
+    }
+
+    fn sign_ecdsa_recoverable(&self, msg: Vec<u8>) -> Result<Vec<u8>, SignerError> {
+        let seed = self.seed();
+        let secp = Secp256k1::new();
+        let keypair = Xpriv::new_master(self.network, &seed)
+            .map_err(|e| anyhow::anyhow!("Could not get signer keypair: {e}"))?
+            .to_keypair(&secp);
+        let s = msg.as_slice();
+
+        let double_hashed_msg: Message = Message::from_digest_slice(s)
```

Comment:
nit: variable name is misleading as we don't know it's double hashed

Reply:
Fixed